crates/chia-bls/src/bls_cache.rs

```diff
@@ -42,13 +41,22 @@ impl BlsCache {
     pub fn is_empty(&self) -> bool {
         self.cache.is_empty()
     }
+    
+    pub fn update(&mut self, new_items: Vec<([u8; 32], GTElement)>) {
```

Comment:
I don't think you should take a vector by value. There's no need to move it in here. In fact, you probably want to take an iterator of `GTElement`, so the python binding can avoid generating an intermediate vector.

Also, as you add this, it probably would make sense to update the python binding to use these functions as well.

https://github.com/Chia-Network/chia_rs/blob/main/crates/chia-bls/src/bls_cache.rs#L139-L167